py/server/tests/test_table_data_service.py

```diff
@@ -421,10 +425,13 @@ def test_partition_size_sub_failure(self):
         table = data_service.make_table(TableKeyImpl("test"), refreshing=True)
 
         # wait for location/size subscription to be established
-        self.wait_ticking_table_update(table, 2, 1)
+        # hold the exclusive lock to ensure table initialization is complete before size failure is triggered
+        with exclusive_lock(table):
+            table = table.coalesce()
 
+        # the test backend will trigger a size subscription failure
+        # if not backend.is_size_sub_failure_cb_called:
         with backend.size_sub_failure_cb_called_cond:
-            # the test backend will trigger a size subscription failure
             if not backend.is_size_sub_failure_cb_called:
```

Comment:
You need to check the `called` boolean after the wait. Conditions can have spurious wakeups. The correct pattern is typically a while loop, with a decreasing wait amount in case you wake up early.